### PR TITLE
Add commands to browse saved songs / albums + current user's playlists

### PR DESCRIPTION
**Pull request by roelofvkr**
# New commands
* `Me` gateway command for stuff related to the user
  * `saved` Browse through tracks or albums you have saved to your library
  * `playlists` Browse through the playlists the user is currently following

# Other new things in app
* Press <kbd>s</kbd> to save the currently selected track or album to your library. Pressing S in the library will remove the song from your library

# Other stuff
* Added makefile
* Added .gitignore

### CODE REVIEW

**Comment by joshuathompson** [comment]
Really cool PR and some great utility that I have been meaning to add eventually with the Makefile. I will review this when I have some time in the next few days. Thanks for the contribution!

**Comment by roelofvkr** [comment]
@joshuathompson Did you get the chance to review it? i'd love to hear feedback on what i wrote

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+os = $(word 1, $@)
+
+# Make for all the platforms
+.PHONY: $(PLATFORMS)
+ifeq ($(OS),Windows_NT)
+$(PLATFORMS):
+	set GOOS=$(os)&& set GOARCH=amd64&& $(GOBUILD) -o $(BINARY_OUTPUT)-$(VERSION)-$(os)-amd64$(if $(filter $(os),windows),.exe,) main.go
+else
+$(PLATFORMS):
+	GOOS=$(os) GOARCH=amd64&& $(GOBUILD)-o $(BINARY_OUTPUT)-$(VERSION)-$(os)-amd64$(if $(filter $(os),windows),.exe,) naub.go
```
Comment (joshuathompson) [inline Makefile]: I think this should follow the convention of the above command, as of now it's missing an initial `set`, had to add a space here `$(GOBUILD)-o` before the `-o` and change `naub.go` to `main.go` in order to get the build to work.

Would be nice to have some build instructions in the README 😄

**Review by joshuathompson — CHANGES_REQUESTED** [review]
https://github.com/joshuathompson/baton/blob/master/api/account.go#L35

On this line we need to add `user-library-modify` in order for those commands to return something other than a 403.

This is a great PR... once these items are addressed I will take another look and immediately merge this 👍

File: Makefile (hunk 1 above)
Comment (roelofvkr) [reply Makefile]: Oops, i'm sorry that must've been a typo i hadn't noticed because I use a windows machine myself

**Comment by roelofvkr** [comment]
I updated the things you pointed out, thanks. I also wrote a few lines in the readme explaining how to build

**Review by joshuathompson — APPROVED** [review]

**Comment by joshuathompson** [comment]
Tested with those changes and can confirm that the build process works on OSX and that saving works correctly for me now.

Thanks for the contribution and sorry I didn't get to this sooner!

**Pull request merged**